### PR TITLE
[FIX] project: handle limit=None on tags `name_search`

### PR DESCRIPTION
**Pull request by pivi-odoo**
## Issue
After https://github.com/odoo-dev/odoo/commit/213b6885312f3da3bf5bff995861758a1afcde76, some custom filters on tags can throw a stacktrace.

## Steps to reproduce
- Install Project
- Project > Custom Filter > Tags contains 'Internal'
- Stacktrace

## Cause
When `limit=None`, which is explicitly set when creating custom filters, the `name_search` crashes when comparing an `int` (the len (ids)) with the limit which is `None`.

## Fix
Elaborate the condition to handle the case when `limit=None`.

## Affected versions
16.0 up to master

## Reference
opw-3510309

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by MissingNoShiny — APPROVED** [review]
LGTM, thanks for the fix :slightly_smiling_face:

File: addons/project/models/project.py [hunk 1]
```diff
@@ -2779,6 +2779,7 @@ def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get
             project_tasks_tags_domain = [('id', 'in', [row[0] for row in self.env.cr.fetchall()])]
             # we apply the args and limit to the ids we've already found
             ids += self.env['project.tags'].search(expression.AND([args, project_tasks_tags_domain]), limit=limit).ids
-        if len(ids) < limit:
-            ids += self.env['project.tags'].search(expression.AND([args, [('id', 'not in', ids)]]), limit=limit - len(ids)).ids
+        if limit is None or len(ids) < limit:
+            limit = limit - len(ids) if limit else limit
```
[the remark this answers is not in the document]
Comment (pivi-odoo) [reply addons/project/models/project.py]: Thanks for pointing that out, I've pushed the second suggestion :)

[the remark this answers is not in the document]
Comment (MissingNoShiny) [reply addons/project/models/project.py]: I think the second one is less explicit than a ternary :frowning_face:

[the remark this answers is not in the document]
Comment (pivi-odoo) [reply addons/project/models/project.py]: I personally don't have a preference, I just want to merge the fix ASAP. :S

**Comment by pivi-odoo** [comment]
Corrected patch to take into account `False` and `0` into account, as those are equivalent to `None` in the `limit` argument in a `search`.

**Review by auon-odoo — APPROVED** [review]
Thanks for the fix!

robodoo r+

**Comment by auon-odoo** [comment]
robodoo r+

**Pull request closed without merging**